### PR TITLE
feat: SetFormula smart constructors

### PR DESCRIPTION
**Pull request by JonathanStarup**
Adding the smart constructors - one of the more painful to review

### CODE REVIEW

**Comment by magnus-madsen** [comment]
> Adding the smart constructors - one of the more painful to review

Ugh yeah. Not sure I will be able to check for correctness. Ill do a cursory look.

**Comment by JonathanStarup** [comment]
not sure anyone can, thats why property testing is felt required 😄

File: main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala [hunk 1]
```diff
@@ -16,8 +16,12 @@
 
 package ca.uwaterloo.flix.language.phase.unification.set
 
+import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.util.{CofiniteIntSet, InternalCompilerException}
+
 import scala.annotation.nowarn
 import scala.collection.immutable.SortedSet
+import scala.collection.mutable.{ListBuffer, Set as MutSet}
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala:24]: The standard convention in Scala and in the compiler is to write:

```
import scala.collection.mutable
mutable.Set.empty etc.
```

**Review by magnus-madsen — COMMENTED** [review]
I just have one comment about style.

**Comment by magnus-madsen** [comment]
What's the code coverage on this with the random tests?

**Comment by JonathanStarup** [comment]
On the Full FastSetUnifcation branch, 100%. But I will also run prop testing again once I can on master before plugging it in

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**